tls/s2n_client_extensions.c

```diff
@@ -65,12 +66,14 @@ int s2n_client_extensions_send(struct s2n_connection *conn, struct s2n_stuffer *
     if (conn->actual_protocol_version == S2N_TLS12) {
         /* The extension header */
         GUARD(s2n_stuffer_write_uint16(out, TLS_EXTENSION_SIGNATURE_ALGORITHMS));
-        GUARD(s2n_stuffer_write_uint16(out, 4));
+        GUARD(s2n_stuffer_write_uint16(out, (sizeof(s2n_preferred_hashes) * 2) + 2));
```

Comment:
nit: It wasn't immediately apparent why we needed to multiply by 2(the hash is only half of the {hash,sign} tuple). A comment could clear up confusion for inexperienced readers :neckbeard: